### PR TITLE
COG-277 switch userlist to dora

### PR DESCRIPTION
**Pull request by kelfink**
PR ready for review and merge.  It forms the beginning of search (by last_name and wide-open) under ElasticSearch / Dora.

### CODE REVIEW

**Comment by kelfink** [comment]
retest this please

**Comment by kelfink** [comment]
retest this please

**Comment by kelfink** [comment]
retest this please

**Comment by kelfink** [comment]
retest this please

**Comment by kelfink** [comment]
retest this please

File: lib/users/user.rb [hunk 1]
```diff
@@ -30,5 +30,16 @@ class User < Dry::Struct
     attribute :verification_passed, Types::Bool.optional
     attribute :racfid, Types::String.optional
     attribute :roles, Types::Array.optional
+    attribute :auth_header, Types::String.optional
+
+    def self.search(query, auth_header)
```
Comment (RamuKammagani-osi) [inline lib/users/user.rb]: Please move this to the user_repository.rb to maintain the current pattern. Lets just keep user.rb as simple dry-struct.

File: lib/users/user.rb [hunk 2]
```diff
@@ -30,5 +30,16 @@ class User < Dry::Struct
...
+      http_search_service = Infrastructure::HttpService.new(base_url)
+      response = http_search_service.post('/dora/users/user/_search', query, auth_header)
+      response.body
+    end
+
+    def self.base_url
+      Rails.configuration.micro_services['base_search_api_url']
```
Comment (RamuKammagani-osi) [inline lib/users/user.rb]: same here

File: spec/lib/users/user_spec.rb [hunk 3]
```diff
@@ -27,5 +27,20 @@ module Users
       it { is_expected.to have_attribute(:verification_passed, Types::Bool.optional) }
       it { is_expected.to have_attribute(:verification_message, Types::String.optional) }
     end
+
+    describe 'search' do
```
Comment (RamuKammagani-osi) [inline spec/lib/users/user_spec.rb]: same here

**Comment by RamuKammagani-osi** [comment]
retest this please

**Pull request merged**